### PR TITLE
docs(contributing): adding contribution documentation for web components

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

No related issue

### Description

This is a similar document to contributing, except for web components
specifically. This has also been added as a link in Storybook.

### Changelog

**New**

- Documentation on contributing to web components

**Changed**

- Various global contribution links in documentation

### CODE REVIEW

**Review by kennylam — APPROVED** [review]
Looks good thanks @jeffchew!

File: packages/web-components/docs/contributing-to-web-components.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Contributing to Carbon for IBM.com Web Components package
+
+> Guidelines for how to make a code contribution to the Web Components package
+
+## Overview
+
+Thank you for your contribution to Carbon for IBM.com! Below are guidelines on
+what some of the things we would be looking for as part of your contribution.
```
Comment (asudoh) [inline packages/web-components/docs/contributing-to-web-components.md:25]: We should also mention more detailed coding conventions: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/docs/coding-conventions.md

Also should we merge [our developer doc](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.12.0/packages/web-components/docs/developer.md) to this, or add a link from here to there?

**Review by asudoh — COMMENTED** [review]
Thank you for adding this @jeffchew!

File: packages/web-components/docs/contributing-to-web-components.md (hunk 1 above)
Comment (jeffchew) [reply packages/web-components/docs/contributing-to-web-components.md:25]: We should link out, I think the developer doc is pretty dense so probably better to have it separate.

Comment (jeffchew) [reply packages/web-components/docs/contributing-to-web-components.md:25]: Sorry, I mistaken your meaning with coding conventions. Let me see if merging these makes sense...

**Pull request merged**